### ISSUE TITLE
Create tests for new vocabulary actions and events

### ISSUE THREAD

**Issue by argiepiano** [backdrop-contrib/rules#151]
While `rules.test` provides some vocabulary tests under testCustomWrapperClasses (providing them as parameters to an action, accessing their properties through the data selector, and setting their properties), it'd be a good idea to add a few more to test the new actions (fetch, create, save and delete) and the new events (presave, insert, update, predelete and delete).

**Comment by argiepiano** [comment]
I just pushed a commit that adds back a functional implementation of vocabulary_unchanged and all its tokens. It's working great on my end. 

What I did is add code to `rules_taxonomy_vocabulary_presave` to store the original vocabulary under `$vocabulary->original`. This is done only if `original` doesn't exist, meaning that if/when the core PR to add it is merged, it will simply not be added by `rules_taxonomy_vocabulary_presave`.

I tested using the test Rule pasted above in this issue - and also checked the tokens provided. All good. Test and let me know if you see anything fishy.
